autoload/go/config.vim

```diff
@@ -240,20 +240,42 @@ function! go#config#SetTemplateAutocreate(value) abort
   let g:go_template_autocreate = a:value
 endfunction
 
+function! go#config#Metalinter() abort
+  return get(g:, "go_metalinter", "gometalinter")
+endfunction
+
 function! go#config#MetalinterCommand() abort
   return get(g:, "go_metalinter_command", "")
 endfunction
 
 function! go#config#MetalinterAutosaveEnabled() abort
-  return get(g:, 'go_metalinter_autosave_enabled', ['vet', 'golint'])
+  let l:default_enabled = ["vet", "golint"]
+
+  if go#config#Metalinter() == "golangci-lint"
+    let l:default_enabled = ["deadcode", "errcheck", "ineffassign", "structcheck", "typecheck", "varcheck"]
+  endif
+
+  return get(g:, "go_metalinter_autosave_enabled", default_enabled)
 endfunction
 
 function! go#config#MetalinterEnabled() abort
-  return get(g:, "go_metalinter_enabled", ['vet', 'golint', 'errcheck'])
+  let l:default_enabled = ["vet", "golint", "errcheck"]
+
+  if go#config#Metalinter() == "golangci-lint"
+    let l:default_enabled = ["deadcode", "errcheck", "ineffassign", "structcheck", "typecheck", "varcheck", "gosimple", "govet", "staticcheck", "unused"]
```

Comment:
A similar question here: why have you chosen these as the defaults?

At a minimum, I'd expect the defaults for golangci-lint to at least use `golint`, `govet`, and `errcheck` to maintain parity with gometalinter.

Reply:
these are just straight from the golangci-lint defaults https://github.com/golangci/golangci-lint#enabled-by-default-linters

Reply:
That makes sense; I'm just considering whether vim-go should try to maintain parity for the default linters regardless of whether it's using `gometalinter` or `golangci-lint` or whether it's ok to vary the default linters depending on which metalinter is being used.